update/v1alpha1/types_update_status_health.go

```diff
@@ -0,0 +1,129 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// HealthInsight is a piece of actionable information produced by an insight producer about the health
+// of the cluster or an update
+type HealthInsight struct {
+	// startedAt is the time when the condition reported by the insight started
+	// +required
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Format=date-time
+	StartedAt metav1.Time `json:"startedAt"`
+
+	// scope is list of objects involved in the insight
+	// +required
+	Scope InsightScope `json:"scope"`
+
+	// impact describes the impact the reported condition has on the cluster or update
+	// +required
+	Impact InsightImpact `json:"impact"`
+
+	// remediation contains information about how to resolve or prevent the reported condition
+	// +required
+	Remediation InsightRemediation `json:"remediation"`
+}
+
+// InsightScope is a list of resources involved in the insight
+type InsightScope struct {
+	// type is either ControlPlane or WorkerPool
+	// +required
+	Type ScopeType `json:"type"`
+
+	// resources is a list of resources involved in the insight, of any group/kind
+	// +optional
+	// +listType=atomic
+	// +kubebuilder:validation:MaxItems=16
+	Resources []ResourceRef `json:"resources,omitempty"`
+}
+
+// ScopeType is one of ControlPlane or WorkerPool
+// +kubebuilder:validation:Enum=ControlPlane;WorkerPool
+type ScopeType string
+
+const (
+	// ControlPlane is used for insights that are related to the control plane (including control plane pool or nodes)
+	ControlPlaneScope ScopeType = "ControlPlane"
+	// WorkerPool is used for insights that are related to a worker pools and nodes (excluding control plane)
+	WorkerPoolScope ScopeType = "WorkerPool"
+)
+
+// InsightImpact describes the impact the reported condition has on the cluster or update
+type InsightImpact struct {
+	// level is the severity of the impact
+	// +required
+	Level InsightImpactLevel `json:"level"`
```

Comment:
What are the expected values for this, and what do they mean? Should be included in the godoc